.travis.yml

```diff
@@ -1,21 +1,39 @@
 language: python
 sudo: required
+
+
 python:
-    - 3.5
-    - 3.6
+  - '3.6'
+  - '3.5'
+matrix:
+  include:
+    - python: '3.6'
+      env: TOKIO="y"
+  allow_failures:
+    - env: TOKIO="y"
 
 install:
     - pip install --upgrade setuptools
     - pip install codecov
     - pip install -Ur requirements-dev.txt
     - python setup.py develop
+    - |
+        export RUST_VERSION=1.17.0
+        python -c "import sysconfig; print('\n'.join(map(repr,sorted(sysconfig.get_config_vars().items()))))"
+        mkdir ~/rust-installer
+        curl -sL https://static.rust-lang.org/rustup.sh -o ~/rust-installer/rustup.sh
+        sh ~/rust-installer/rustup.sh --prefix=~/rust --spec=$RUST_VERSION -y --disable-sudo
+        export PATH="$HOME/rust/bin:$PATH"
+        export PYTHON_LIB=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDIR'))")
+        find $PYTHON_LIB
+        export LIBRARY_PATH="$LIBRARY_PATH:$PYTHON_LIB"
+        export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$PYTHON_LIB:$HOME/rust/lib"
+        rustc -V
+        if [ "$TOKIO" = "y" ]; then
```

Comment:
Does it worth to install rust all the time when TOKIO is `n`? Seems like a waste of a valuable build time.